### PR TITLE
Directly authenticate when building with password

### PR DESCRIPTION
**Pull request by damien-carcel**
When building a client with `AkeneoPimClientBuilder::buildAuthenticatedByPassword`, the returned client has no access nor refresh tokens. Both corresponding getters will simply return `null`.
For the token to be set, one have to perform at least one API call, so the lower level `AuthenticatedHttpClient::sendRequest()` method tries to get the access token and authenticate if there is none.

From an integrator POV, this is confusing, as one would expect to get the tokens right away to store them for future usage.

This PR proposes to call the authentication API when building a client with username and password. This way, the returned client already contains an access token and a refresh token.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-DOCKER_RUN = DOCKER_BUILDKIT=1 docker-compose run php_client
+DOCKER_RUN ?= DOCKER_BUILDKIT=1 docker-compose run php_client
```
Comment (damien-carcel) [inline Makefile:1]: This allows to override locally the environment variable with an export. In my case, I don't use `docker-compose` but `docker compose`.

Comment (LevFlavien) [reply Makefile:1]: @samokiss your solution to be able to use Compose V2 :slightly_smiling_face:

Comment (damien-carcel) [reply Makefile:1]: Another solution would be to replicate what JM did on the PIM (I think the PR is not merged yet) and what we did for now quite some time on Shared Catalogs: abandon Compose v1. Compose v2 is out for more than 2 years now, and Circle CI latests "machine" images work like a charm with it.

Comment (tomglvng) [reply Makefile:1]: Thank you, we have this problem on our project! And...--> https://docs.docker.com/compose/release-notes/#:~:text=From%20the%20end%20of%20June,from%20all%20Docker%20Desktop%20versions.

**Review by tomglvng — APPROVED** [review]

**Review by val-- — APPROVED** [review]

**Pull request merged**